core/src/main/java/org/elasticsearch/cluster/routing/allocation/RerouteExplanation.java

```diff
@@ -52,23 +52,21 @@ public Decision decisions() {
     }
 
     public static RerouteExplanation readFrom(StreamInput in) throws IOException {
-        String commandName = in.readString();
-        AllocationCommand command = AllocationCommands.lookupFactorySafe(commandName).readFrom(in);
+        AllocationCommand command = in.readAllocationCommand();
         Decision decisions = Decision.readFrom(in);
         return new RerouteExplanation(command, decisions);
     }
 
     public static void writeTo(RerouteExplanation explanation, StreamOutput out) throws IOException {
-        out.writeString(explanation.command.name());
-        AllocationCommands.lookupFactorySafe(explanation.command.name()).writeTo(explanation.command, out);
+        out.writeAllocationCommand(explanation.command);
```

Comment:
if we need to have this discussion I'd suggest to have it on a separate issue and not hold this PR because of this.